src/language/nodes.py

```diff
@@ -234,11 +234,15 @@ def member_typename(self):
 
         return type_name
 
+    @property
+    def _is_member_type_wrapped_as_shared_pointer(self):
+        return not (self.is_vector or self.is_base_type_node or self.is_ptr_excluded_node)
+
     @property
     def member_rvalue_typename(self):
         """returns rvalue reference type when used as returned or parameter type"""
         typename = self.member_typename
-        if not self.is_integral_type_node:
+        if not self.is_integral_type_node and not self._is_member_type_wrapped_as_shared_pointer:
```

Comment:
Just a nitpick: the check for « is_integral_type_node«  is not needed because we check for « is_base_type_node«  and that includes all integral types node?

Reply:
I'm not sure I understand and there's too many `not`s for my brain. The logic is: We don't add `const&` to integral types and anything wrapped in a shared pointer. The first condition makes the first half of the sentence clear; and the second condition the second half. (For that reason alone I wouldn't golf this.)

Can it be golfed? I don't think so because an `int` is not wrapped in a shared pointer and hence  `_is_member_type_wrapped_as_shared_pointer` should be `False`.

Let's go slowly for the example of an `int` and check if it will be wrapped if the condition is only:
```
if not self._is_member_type_wrapped_as_shared_pointer:
   return "const " + typename + "&"
```
Let's start:
```
is_base_type_node == True
=> _is_member_type_wrapped_as_shared_pointer == False
=> not _is_member_type_wrapped_as_shared_pointer == True
=> add `const &` to the return type.
```

Reply:
my bad! missed the second `not` in `_is_member_type_wrapped_as_shared_pointer`!